app/src/org/commcare/dalvik/activities/SingleAppManagerActivity.java

```diff
@@ -0,0 +1,285 @@
+        if (isArchived) {
+            archiveButton.setText("Unarchive");
+        } else {
+            archiveButton.setText("Archive");
+        }
+    }
+    
+    @Override
+    public void onResume() {
+        super.onResume();
+        refresh();
+    }
+    
+    private void refresh() {    
+        //refresh old profile warning 
+        TextView warning = (TextView) findViewById(R.id.profile_warning);
+        if (appRecord.fromOldProfileFile()) {
+            warning.setVisibility(View.VISIBLE);
+        } else {
+            warning.setVisibility(View.GONE);
+        }
+        
+        //refresh validate button
+        Button validateButton = (Button) findViewById(R.id.verify_button);
+        if (appRecord.resourcesValidated()) {
+            validateButton.setVisibility(View.INVISIBLE);
+        } else {
+            validateButton.setVisibility(View.VISIBLE);
+        }
+    }
+    
+    @Override
+    protected void onActivityResult(int requestCode, int resultCode, Intent intent) {
+        switch (requestCode) {
+        case CommCareHomeActivity.UPGRADE_APP:
+            if(resultCode == RESULT_CANCELED) {
+                Toast.makeText(this, "Your update did not complete", Toast.LENGTH_LONG).show();
+            } else if(resultCode == RESULT_OK) {
+                //Set flag that we should autoupdate on next login
+                SharedPreferences preferences = CommCareApplication._().getCurrentApp().getAppPreferences();
+                preferences.edit().putBoolean(CommCarePreferences.AUTO_TRIGGER_UPDATE,true);
+            }
+            break;
+        case CommCareHomeActivity.MISSING_MEDIA_ACTIVITY:
+            refresh();
+            if (resultCode == RESULT_CANCELED) {
+                AlertDialog.Builder builder = new AlertDialog.Builder(this);
+                builder.setTitle("Media Not Verified");
+                builder.setMessage(R.string.skipped_verification_warning_2)
+                    .setPositiveButton("OK", new DialogInterface.OnClickListener() {
+
+                            @Override
+                            public void onClick(DialogInterface dialog,
+                                    int which) {
+                                dialog.dismiss();
+                            }
+                            
+                        });
+                dialog = builder.create();
+                dialog.show();
+            }
+            else if (resultCode == RESULT_OK) {
+                Toast.makeText(this, "Media Validated!", Toast.LENGTH_LONG).show();
+            }
+            break;
+        case CommCareHomeActivity.RESTART_APP:
+            if (dialog != null) {
+                dialog.dismiss();
+            }
+            Intent i = new Intent(getApplicationContext(), AppManagerActivity.class);
+            startActivity(i);
+        }
+    }
+    
+    /** Uninstalls the selected app **/
+    public void uninstall() {
+        CommCareApplication._().logout();
```

Comment:
do we have a good way of attempting to try to make these database ops atomic or trying to make sure it's as clean as possible? IE: The best way to do this would probably be to set the resource state (or some variable) to "STATE_DELETE_REQUESTED", try the delete, and then remote the record entirely. In this state it's fairly easy for some of the actions to complete but not others in a way that oculd leave the app in a really bad state (We don't even set the uninstalled flag until the very, very end, so if we get 2/3 through the app will just be 2/3 deleted and won't know why.